include/turtle/detail/expectation_template.hpp

```diff
@@ -196,16 +221,41 @@ namespace detail
         {
             return invocation_->verify();
         }
-
+#if defined(MOCK_THREAD_SAFE)
+        bool verify(const boost::shared_ptr< condition_variable> &cv, lock &lk) const
+        {
+            if (timeout_)
+            {
+                while (!invocation_->verify())
```

Comment:
It didn't have time to work on this earlier. But here is an answer: it also works. As proof you can look in the test_async.cpp file for a demonstration of this.

Every expectation can have a time out value associated with it. The delaying of the execution is done inside the expectation verification.